core/primitives/src/subtrie.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Child trie related struct
+
+use parity_codec::{Encode, Decode};
+use rstd::prelude::*;
+#[cfg(feature = "std")]
+use serde_derive::{Serialize, Deserialize};
+
+#[cfg(feature = "std")]
+pub use impl_serde::serialize as bytes;
+
+/// keyspace type.
+pub type KeySpace = Vec<u8>;
+
+
+/// key of subtrie in parent trie.
+pub type ParentTrie = Vec<u8>;
+
```

Comment:
This scheme does not work with multilevel of child trie (long keyvalue db key at some point should not be suitable depending of the database use), that is on point that makes me think that 'keyspace' could be transmit to keyvaluedb api and not be mix with prefix (intent of 'keyspace' is very different from 'prefix' intent)